CMakeLists.txt

```diff
@@ -273,6 +273,33 @@ endif(BUILD_WITH_AUTODIFF_SUPPORT)
 
 if(BUILD_WITH_CASADI_SUPPORT)
   add_project_dependency(casadi 3.4.5 REQUIRED PKG_CONFIG_REQUIRES "casadi >= 3.4.5")
+
+  # Normalize CasADi link target across versions.
+  # Prefer imported target (casadi::casadi), then legacy target (casadi), then library variables.
+  if(TARGET casadi::casadi)
+    set(CASADI_TARGET casadi::casadi)
+  elseif(TARGET casadi)
+    set(CASADI_TARGET casadi)
+  elseif(DEFINED CASADI_LIBRARIES)
+    set(CASADI_TARGET ${CASADI_LIBRARIES})
+  elseif(DEFINED casadi_LIBRARIES)
+    set(CASADI_TARGET ${casadi_LIBRARIES})
```

Comment:
During our discussion, we weren’t able to identify a CasADI CMake module that lacked a defined target. Could you explain why you’re aiming to manage this specific case?

```suggestion
```